src/utils/reward.ts

```diff
@@ -0,0 +1,123 @@
+    query {
+        feeStats(first:${Math.floor(sampleDays)}) {
+            marginAndLiquidation
+            swap
+            mint
+            burn
+        }
+        hlpStats(first:1, orderBy:timestamp, orderDirection:desc) {
+            aumInUsdg
+        }
+    }`;
+
+export const fetchFeeApr = async (
+  network: Network,
+  overrideAumInUsdg?: BigNumber,
+  sampleDays = 7
+) => {
+  if (network !== "arbitrum")
+    throw new Error("Cannot fetch if not on arbitrum");
+  const response = await request<GraphResponse>(
+    config.theGraphEndpoints[network].trade,
+    getQuery(sampleDays)
+  );
+  if (response.feeStats.length === 0) return 0;
+  if (response.hlpStats.length === 0) return 0;
+
+  const total = response.feeStats.reduce(
+    (acc, curr) =>
+      acc
+        .add(curr.marginAndLiquidation)
+        .add(curr.swap)
+        .add(curr.mint)
+        .add(curr.burn),
+    ethers.constants.Zero
+  );
+
+  const meanDailyFee =
+    +ethers.utils.formatUnits(total, PRICE_DECIMALS) / response.feeStats.length;
+  const yearlyFee = meanDailyFee * 365;
+  // usdg has 18 decimals
+  const aum = +ethers.utils.formatUnits(
+    overrideAumInUsdg || response.hlpStats[0].aumInUsdg,
+    18
+  );
+
+  return yearlyFee / aum;
+};
+
+export const fetchAveragePoolBoost = async (
+  poolId: BigNumberish,
+  provider: ethers.providers.Provider,
+  network: Network
+) => {
+  const address = config.protocol[network]?.protocol.rewardPool;
+  if (!address) throw new Error(`No reward pool on ${network}`);
+  const rewardPool = RewardPool__factory.connect(address, provider);
+  const pool = await rewardPool.getPool(poolId);
+  const averageBoost = pool.totalDeposits
+    .mul(BASIS_POINTS_DIVISOR)
+    .div(pool.totalRealDeposits);
+  return +averageBoost / BASIS_POINTS_DIVISOR;
+};
+
+export const fetchUserBoost = async (
+  user: string,
+  poolId: BigNumberish,
+  provider: ethers.providers.Provider,
+  network: Network
+) => {
+  const rewardPoolAddress = config.protocol[network]?.protocol.rewardPool;
+  const governanceLockAddress =
+    config.protocol[network]?.protocol.governanceLock;
+  if (!rewardPoolAddress || !governanceLockAddress)
+    throw new Error(`No protocol on ${network}`);
+  const rewardPool = RewardPool__factory.connect(rewardPoolAddress, provider);
+  const gov = GovernanceLock__factory.connect(governanceLockAddress, provider);
+  const [deposit, pool, totalSupply] = await Promise.all([
+    rewardPool.getDeposit(user, poolId),
+    rewardPool.getPool(poolId),
+    gov.totalSupply(),
+  ]);
+
+  const boostDepositAmount = deposit.boostWeight
+    .mul(pool.totalRealDeposits)
+    .mul(15)
+    .div(totalSupply)
+    .div(10);
+  const boost = deposit.amount.add(boostDepositAmount);
+  let boostRatio = boost.mul(BASIS_POINTS_DIVISOR).div(deposit.amount);
+  if (boostRatio.gt(2.5 * BASIS_POINTS_DIVISOR)) {
+    boostRatio = BigNumber.from(2.5 * BASIS_POINTS_DIVISOR);
+  }
+
+  return 1 + +boostRatio / BASIS_POINTS_DIVISOR;
```

Comment:
Boost ratio is up to 2.5, not 1.5, so we don't need to add 1 here
```suggestion
  return +boostRatio / BASIS_POINTS_DIVISOR;
```